libreoffice-core/vcl/source/outdev/font.cxx

```diff
@@ -384,6 +387,49 @@ void OutputDevice::BeginFontSubstitution()
     pSVData->maGDIData.mbFontSubChanged = false;
 }
 
+void OutputDevice::AddCustomMacroFonts()
+{
+    OutputDevice* pDevice = Application::GetDefaultDevice();
+    pDevice->ImplAddCustomMacroFonts();
+}
+
+void OutputDevice::ImplAddCustomMacroFonts()
+{
+    std::list<std::string> fontNames = {
+        "Carlito",
+        "Carlito-Bold",
+        "Carlito-Italic",
+        "Carlito-BoldItalic",
+        "Caladea",
+        "Caladea-Bold",
+        "Caladea-Italic",
+        "Caladea-BoldItalic"
+    };
+
+    OUString installationRootPath(
+        getOfficePath(psp::whichOfficePath::InstallationRootPath)
+    );
+    std::string extension = ".ttf";
+
+    OUStringBuffer basePathBuffer(256);
+    basePathBuffer.appendAscii("file://");
+    basePathBuffer.append(installationRootPath);
+    basePathBuffer.appendAscii("/");
+    basePathBuffer.append(LIBO_SHARE_RESOURCE_FOLDER);
+    basePathBuffer.append("/macro_fonts/");
+    OUString basePath = basePathBuffer.makeStringAndClear();
```

Comment:
More idiomatic and cross-platform guaranteed version. `psp::` is technically only available on unix platforms, I hacked parts of it together to just barely work with Windows.

```suggestion
    OUString basePath("$BRAND_BASE_DIR/" LIBO_SHARE_RESOURCE_FOLDER "/macro_fonts/");
    rtl::Bootstrap::expandMacros(basePath);
```